lib/ui/painting.dart

```diff
@@ -1577,6 +1577,15 @@ class Codec extends NativeFieldWrapperClass2 {
   /// * -1 for infinity repetitions.
   int get repetitionCount native 'Codec_repetitionCount';
 
+  /// Clears the in-memory decoded frame cache and disables future caching of
+  /// decoded frames. This negatively impacts CPU usage, but reduces memory
+  /// consumption.
+  Future<bool> clearAndDisableFrameCache() => _futurize(_clearAndDisableFrameCache);
```

Comment:
"And" is usually a bad sign in an API name -- what if you want to do one but not the other?

It's not clear to me why clearing the cache and disabling the cache are related.

Reply:
I agree in general, but the functionality is pretty tightly coupled here for the specific usecase this PR is addressing. If somebody disabled the cache without clearing it, they'd end up holding on to all the previously decoded frames in memory still. If somebody cleared the cache but didn't disable it, they'd immediately start caching the same frames over again with the next `getNextFrame` call.  I'm not seeing a reason somebody would want to do either, but maybe there's a use case I'm not thinking of? Or are you thinking it's better to always keep the discrete functionality like this isolated, just in case?

Reply:
Presumably clearing the cache for animated GIFs that aren't currently on the screen won't cause them to repopulate the cache quickly, since they're not animating, generally.

It seems to me that reacting to memory pressure events by disabling the cache for every image (permanently, since we have no signal by which we can turn them back on) is just going to mean that Flutter apps will go from using a lot of memory to being slow in a sudden performance cliff, which isn't a great user experience.

Maybe we want a different metric? e.g. something based on how long it takes to go around the loop, or how much memory the cache uses, or some such?